### DISCUSSION TITLE
Using via version and bukkit api with Minestom

### DISCUSSION THREAD

**Discussion by Outfit8TSB** [Minestom/Minestom#2269] [General]
Smart Tips:
> [!CAUTION]
> This Tip is for groovy gradle

you can make your Server Support via version By slapping
`maven { url 'https://repo.viaversion.com' }` in repository block and
```
Implementation 'com.viaversion:viaversion:5.0.0-SNAPSHOT'
Implementation 'com.viaversion:viabackwards:5.0.0-SNAPSHOT'
Implementation 'com.viaversion:viarewind:3.2.0-SNAPSHOT'
```
On Dependency Block. this way, you server will support 1.7-1.21.
If You Want to Use Bukkit API, you can do it by adding Bukkit API Dependency.
`maven { url '[BUKKIT API PROVIDER REPOSITORY]' }` in repository block
and 
`Implementation '[BUKKIT API]'`
Replace [BUKKIT API] with your desired Bukkit API Fork and Replace [BUKKIT API PROVIDER REPOSITORY] with Bukkit API Fork Provider.
Example: If You Want Paper API for 1.20.6:
`maven { url 'https://repo.papermc.io/repository/maven-releases/' }` in repository block
and 
`Implementation 'io.papermc:paper-api:1.20.6-R0.1-SNAPSHOT'` in dependency block
if you want to use Sponge API, you can do it as:
`maven { url '[SPONGE API PROVIDER REPOSITORY]' }` (ordinary, we use https://repo.spongepowered.org/repository/maven-public unless you have your own fork)
and
`Implementation '[SPONGEAPI DEPENDENCY]'` (ordinary, we use org.spongepowered:spongeapi:12.0.0-SNAPSHOT)
and then you can use sponge in your dependency!

so, if you are developer level, you can use `build.gradle` in your minestom server project to turn your minestom server into a mecca of Vanilla Minecraft Plugin APIs and old and outdated minecraft versions.
> Ned Flanders: build.gradle? mecca?

**Comment by FrederikHeinrich** [comment]
and how to implement it?

**Reply by Outfit8TSB** [reply]
You Can Learn from many extensions making bukkit api to run in minestom How to implement the bukkit api, and boom.
Same goes to Sponge.
To Run ViaVersion~Rewind, You Need ViaLoader.
To Use it with Geyser-Standalone, Learn from Geyser-Utils to implement floodgate, and you are done.

**Reply by Outfit8TSB** [reply]
if you are confused, here is the java snippet. (Extensions: Bukstom)
```
org.MultiNet.Multibukkit.start.enablebukkit()
    public void enablebukkit() {
        Bukkit.setServer(server);
        server.loadPlugins();
        server.enablePlugins(PluginLoadOrder.STARTUP);
        server.enablePlugins(PluginLoadOrder.POSTWORLD); // TODO actually post world

        System.out.Println("Bukkit Plugins Loaded, Initializing Ores...");
    }
```
i converted the kotlin using copilot

**Reply by FrederikHeinrich** [reply]
i´m not confused :) i just dont want to use BukkitAPI anymore :)
Sponge is okay for Client modding... but for the server side.... i will hopefully never touch Bukkit/Paper/Velocity etc. again. :)

**Reply by Outfit8TSB** [reply]
okay.

**Reply by Obscrts** [reply]
How would you just implement ViaVersion and ViaBackwards for compatibility in Minestom so older Versions can join as well?

**Reply by Outfit8TSB** [reply]
you need ViaLoader to that. but also, you can learn from ViaStom(Minestom Implementation) to do that.

**Reply by Outfit8TSB** [reply]
but is it illegal to implement Bukkit API and ViaVersion without credit?

**Reply by Obscrts** [reply]
Where would I find a repo of ViaStom?

**Reply by Outfit8TSB** [reply]
https://github.com/Stredox02/ViaStom

**Reply by Outfit8TSB** [reply]
Also, ViaLoader does loads it.

**Comment by Outfit8TSB** [comment]
i guess i am working on a mecca of old versions and bukkit and spongewith using build.gradle, with 
build.gradle, i can make the minestom work with Bukkit/Sponge plugins so i can overtake glowstonemc. muhahahahahaha. >:)

**Comment by slimeMMVI** [comment]
this is ragebait right?

**Reply by Outfit8TSB** [reply]
nope.
